demos/rag_agentic/notebooks/Level5_agents_and_mcp.ipynb

```diff
@@ -200,9 +200,183 @@
+      "\n",
+      "```yaml\n",
+      "apiVersion: v1\n",
+      "kind: Pod\n",
+      "metadata:\n",
+      "  annotations:\n",
+      "    openshift.io/scc: restricted-v2\n",
+      "    seccomp.security.alpha.kubernetes.io/pod: runtime/default\n",
+      "  creationTimestamp: \"2025-05-13T23:12:36Z\"\n",
+      "  labels:\n",
+      "    app.kubernetes.io/component: slack-test\n",
+      "    app.kubernetes.io/managed-by: kubernetes-mcp-server\n",
+      "    app.kubernetes.io/name: slack-test\n",
+      "    app.kubernetes.io/part-of: kubernetes-mcp-server-run-sandbox\n",
+      "  name: slack-test\n",
+      "  namespace: llama-serve\n",
+      "  resourceVersion: \"935457369\"\n",
+      "  uid: d4495d95-8ff1-4bd5-a03e-40bfb3b88f6b\n",
+      "spec:\n",
+      "  containers:\n",
+      "  - image: docker.io/hemaveeradhi/error-test-pod:latest\n",
+      "    imagePullPolicy: Always\n",
+      "    name: slack-test\n",
+      "    resources: {}\n",
+      "    securityContext:\n",
+      "      allowPrivilegeEscalation: false\n",
+      "      capabilities:\n",
+      "        drop:\n",
+      "        - ALL\n",
+      "      runAsNonRoot: true\n",
+      "      runAsUser: 1001770000\n",
+      "    terminationMessagePath: /dev/termination-log\n",
+      "    terminationMessagePolicy: File\n",
+      "    volumeMounts:\n",
+      "    - mountPath: /var/run/secrets/kubernetes.io/serviceaccount\n",
+      "      name: kube-api-access-7qxdd\n",
+      "      readOnly: true\n",
+      "  dnsPolicy: ClusterFirst\n",
+      "  enableServiceLinks: true\n",
+      "  imagePullSecrets:\n",
+      "  - name: default-dockercfg-bxwvb\n",
+      "  preemptionPolicy: PreemptLowerPriority\n",
+      "  priority: 0\n",
+      "  restartPolicy: Always\n",
+      "  schedulerName: default-scheduler\n",
+      "  securityContext:\n",
+      "    fsGroup: 1001770000\n",
+      "    seLinuxOptions:\n",
+      "      level: s0:c42,c24\n",
+      "    seccompProfile:\n",
+      "      type: RuntimeDefault\n",
+      "  serviceAccount: default\n",
+      "  serviceAccountName: default\n",
+      "  terminationGracePeriodSeconds\n",
+      "\u001b[0m\n",
+      "========== Query processing completed ========== \n",
+      "\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Create simple agent with tools\n",
+    "agent = Agent(\n",
+    "    client,\n",
+    "    model= model_id,  # replace this with model_id to get the value of INFERENCE_MODEL_ID environment variable\n",
+    "    instructions = model_prompt , # update system prompt based on the model you are using\n",
+    "    tools=[\"mcp::openshift\"],\n",
+    "    tool_config={\"tool_choice\":\"auto\"},\n",
+    "    sampling_params=sampling_params\n",
+    ")\n",
+    "\n",
+    "user_prompts = [\"Create a pod called slack-test in the llama-serve namespace using the docker.io/hemaveeradhi/error-test-pod:latest image\"]\n",
```

Comment:
Unless we need to test this, I think we should wait on merging until we have a `quay.io/redhat-et/` image to use here.

Reply:
quay.io is now working, pushed image to `quay.io/redhat-et/failing-test-pod`